lightning/src/util/test_utils.rs

```diff
@@ -264,6 +265,157 @@ impl<'a> chain::Watch<EnforcingSigner> for TestChainMonitor<'a> {
+	JusticeTxFormed(Transaction),
+}
+
+pub struct WatchtowerPersister {
+	pub persister: TestPersister,
+	pub destination_script: Script,
+	pub channel_watchtower_state: Mutex<HashMap<OutPoint, HashMap<u64, WatchtowerState>>>,
+}
+
+impl WatchtowerPersister {
+	pub(crate) fn new(destination_script: Script) -> Self {
+		WatchtowerPersister {
+			persister: TestPersister::new(),
+			destination_script,
+			channel_watchtower_state: Mutex::new(HashMap::new()),
+		}
+	}
+
+	fn build_justice_tx(&self, commitment_txid: Txid, output_idx: u32, value: u64) -> Transaction {
+		let mut justice_tx = Transaction {
+			version: 2,
+			lock_time: bitcoin::PackedLockTime::ZERO,
+			input: vec![TxIn {
+				previous_output: bitcoin::OutPoint {
+					txid: commitment_txid,
+					vout: output_idx,
+				},
+				script_sig: Script::new(),
+				sequence: Sequence::ENABLE_RBF_NO_LOCKTIME,
+				witness: Witness::new(),
+			}],
+			output: vec![TxOut {
+				script_pubkey: self.destination_script.clone(),
+				value,
+			}],
+		};
+		// Naive fee calculation to pass tests
+		let min_fee = (justice_tx.weight() as u64 + 3) / 4;
+		justice_tx.output[0].value -= min_fee * 2;
+		justice_tx
+	}
+
+	fn sign_justice_tx<Signer: sign::WriteableEcdsaChannelSigner>(&self, mut justice_tx: Transaction, secret: &[u8; 32], value: u64, data: &channelmonitor::ChannelMonitor<Signer>) -> Result<Transaction, ()> {
+		let secp_ctx = Secp256k1::new();
+		let per_commitment_key = SecretKey::from_slice(secret).unwrap();
+		let per_commitment_point = PublicKey::from_secret_key(&secp_ctx, &per_commitment_key);
+		let revokeable_redeemscript = data.get_revokeable_redeemscript(per_commitment_point);
+
+		let input_idx = 0;
+		let sig = match data.sign_justice_revoked_output(&justice_tx, input_idx, value, &per_commitment_key) {
+			Ok(sig) => sig,
+			Err(_) => return Err(()),
+		};
+
+		let mut ser_sig = sig.serialize_der().to_vec();
+		ser_sig.push(EcdsaSighashType::All as u8);
+		justice_tx.input[0].witness.push(ser_sig);
+		justice_tx.input[0].witness.push(vec!(1));
+		justice_tx.input[0].witness.push(revokeable_redeemscript.clone().into_bytes());
+
+		Ok(justice_tx)
+	}
+}
+
+impl<Signer: sign::WriteableEcdsaChannelSigner> chainmonitor::Persist<Signer> for WatchtowerPersister {
+	fn persist_new_channel(&self, funding_txo: OutPoint, data: &channelmonitor::ChannelMonitor<Signer>, id: MonitorUpdateId) -> chain::ChannelMonitorUpdateStatus {
+		assert!(self.channel_watchtower_state.lock().unwrap()
+			.insert(funding_txo, HashMap::new()).is_none());
+
+		// No ChannelMonitorUpdateStep::LatestCounterpartyCommitmentTxInfo is persisted on channel creation
+		// so we get the same info and update the watchtower
+		let commitment_number = (1u64 << 48) - 1;
+		let output_idx = 0;
+		let commitment_txid = data.get_current_counterparty_commitment_txid().expect("Should have at least one counterparty commitment tx");
+		let value = data.initial_to_counterparty_value();
```

Comment:
Something important I've learned so far about my current approach: the very first counterparty commitment transaction when we open a channel is provided to the `ChannelMonitor` upon its creation, which means we don't get a `ChannelMonitorUpdateStep::LatestCounterpartyCommitmentTxInfo` in the `Persist` implementation for the first commitment tx, which is needed to form the justice tx. Right now I expose some extra stuff just for this case in a fairly hacky way. It may be worth trying to refactor to allow this initial update to go through persistence if it's thought that this sort of persistence API should be better supported, but generally this is making me lean towards exploring using what's already offered to the signer.